### PR TITLE
Adds secret-handshake exercise

### PR DESCRIPTION
**Pull request by amscotti**
Hi, this PR is to add a new exercise for `secret-handshake`, [here is a link to the problem-specifications](https://github.com/exercism/problem-specifications/tree/master/exercises/secret-handshake)

I'm still learning Dart, so feel free to point out anything that needs to be change or improved. I'm also happy to change the `config.json` to better fit your layout for the track.

Thanks! :)

### CODE REVIEW

File: exercises/secret-handshake/lib/example.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class SecretHandshake {
+  final signal = ["wink", "double blink", "close your eyes", "jump"];
+  final reverse = 4;
```
Comment (Stargator) [inline exercises/secret-handshake/lib/example.dart]: `reverse` and `signal` should be private, but I can take it as is.

File: config.json [hunk 2]
```diff
@@ -381,6 +381,18 @@
         "sorting",
         "trees"
       ]
+    },
+    {
+      "slug": "secret-handshake",
+      "uuid": "0687be19-99e8-4b85-b3da-6b91a10acaa3",
+      "core": false,
+      "unlocked_by": null,
```
[the remark this answers is not in the document]
Comment (Stargator) [reply config.json]: @amscotti This can be reverted and unlocked by scrabble-score

[the remark this answers is not in the document]
Comment (amscotti) [reply config.json]: @Stargator Updated

**Comment by amscotti** [comment]
Updated and rebased with master.

**Pull request merged**